cluster_management/src/main/java/com/pinterest/rocksplicator/config/PollingFileWatcher.java

```diff
@@ -121,6 +123,21 @@ public static PollingFileWatcher defaultPerSecondWatcher() {
     return HIGH_RESOLUTION_INSTANCE;
   }
 
+  /**
+   * Creates the default ConfigFileWatcher instance on demand.
+   */
+  public static PollingFileWatcher defaultExpensiveHighResolutionWatcher() {
+    if (EXPENSIVE_HIGH_RESOLUTION_INSTANCE == null) {
```

Comment:
How is `EXPENSIVE_HIGH_RESOLUTION_INSTANCE` different? More frequent polling?

Reply:
Yes... This is only used in testing... not in prod code. In the test, it is to ensure we get faster notification when the file is downloaded from zk... In prod we don't use this...